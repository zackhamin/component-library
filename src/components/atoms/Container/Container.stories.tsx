import { Container, ContainerProps } from './Container';

export default {
	title: 'atoms/Container',
	component: Container,
};

export const ContainerStory = (args: ContainerProps) => <Container {...args} />;

ContainerStory.storyName = 'Container';
ContainerStory.args = {
	children: (
		<p>
			Adipisicing occaecat fugiat labore exercitation consectetur eu laboris
			aliquip fugiat. Incididunt dolor ipsum sint reprehenderit sunt quis ut
			ipsum Lorem. Amet labore laboris sit sint magna do. Reprehenderit
			consequat culpa veniam laboris adipisicing ut qui nostrud dolor elit.
		</p>
	),
};
ContainerStory.argTypes = {
	centered: {
		control: 'boolean',
	},
	marginTop: {
		control: 'text',
	},
	marginBottom: {
		control: 'text',
	},
};
