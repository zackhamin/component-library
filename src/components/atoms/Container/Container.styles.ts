import { Size } from '@Types/css';
import styled from 'styled-components';

const StyledContainer = styled.div<{
	centered: boolean;
	marginTop: Size;
	marginBottom: Size;
	width: Size;
}>`
	width: ${({ width }) => width};
	max-width: 100%;
	margin: ${({ marginTop, marginBottom, centered }) =>
		`${marginTop} ${centered ? 'auto' : '0'} ${marginBottom}`};
`;

export { StyledContainer };
