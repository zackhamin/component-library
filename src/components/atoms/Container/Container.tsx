import { ReactNode } from 'react';
import { Size } from '@Types/css';
import { StyledContainer } from './Container.styles';

export interface ContainerProps {
	children: ReactNode;
	centered?: boolean;
	marginTop?: Size;
	marginBottom?: Size;
	width?: Size;
}

const Container = ({
	children,
	centered = true,
	marginTop = '0',
	marginBottom = '0',
	width = '114rem',
}: ContainerProps) => {
	return (
		<StyledContainer
			centered={centered}
			marginTop={marginTop}
			marginBottom={marginBottom}
			width={width}
		>
			{children}
		</StyledContainer>
	);
};

export { Container };
