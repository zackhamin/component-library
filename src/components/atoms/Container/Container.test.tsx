import { Container } from '../Container';
import { render } from '@testing-library/react';

describe('Container', () => {
	it('renders correctly', () => {
		const { container } = render(
			<Container>
				<p>
					Adipisicing occaecat fugiat labore exercitation consectetur eu laboris
					aliquip fugiat. Incididunt dolor ipsum sint reprehenderit sunt quis ut
					ipsum Lorem. Amet labore laboris sit sint magna do. Reprehenderit
					consequat culpa veniam laboris adipisicing ut qui nostrud dolor elit.
				</p>
			</Container>
		);
		expect(container.firstChild).toMatchSnapshot();
	});
});
