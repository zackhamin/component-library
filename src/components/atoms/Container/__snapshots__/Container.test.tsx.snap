// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders correctly 1`] = `
.c0 {
  width: 114rem;
  max-width: 100%;
  margin: 0 auto 0;
}

<div
  class="c0"
  width="114rem"
>
  <p>
    Adipisicing occaecat fugiat labore exercitation consectetur eu laboris aliquip fugiat. Incididunt dolor ipsum sint reprehenderit sunt quis ut ipsum Lorem. Amet labore laboris sit sint magna do. Reprehenderit consequat culpa veniam laboris adipisicing ut qui nostrud dolor elit.
  </p>
</div>
`;
